jukebox/src/lib.rs

```diff
@@ -122,31 +138,75 @@ impl Jukebox {
                         },
                     };
 
-                    // Parse the bytes as an Hps
-                    let hps: Hps = match copy_bytes_from_file(&mut iso, real_hps_offset, hps_length)?.try_into() {
-                        Ok(hps) => hps,
-                        Err(e) => {
-                            tracing::error!(target: Log::Jukebox, error = ?e, "Failed to parse bytes into an Hps. Cannot play song.");
-                            continue;
-                        },
-                    };
+                    // Try finding custom song
+                    let mut custom_song_path = None;
+                    if let Some(stage) = hps_to_stage(real_hps_offset) {
+                        let stage_dir = Path::new(&jukebox_path).join(stage);
+                        if let Ok(entries) = read_dir(&stage_dir) {
+                            // Get all files in folder
+                            let files: Vec<_> = entries
+                                .filter_map(|entry| {
+                                    if let Ok(entry) = entry {
+                                        let path = entry.path();
+                                        if path.is_file() {
+                                            if let Some(extension) = path.extension().and_then(|extension| extension.to_str()) {
+                                                if ["mp3", "wav", "ogg", "flac"].contains(&extension.to_lowercase().as_str()) {
+                                                    return Some(path);
+                                                }
+                                            }
+                                        }
+                                    }
+                                    None
+                                })
```

Comment:
This can be simplified using `?` early returns for a less nested structure:
```rs
.filter_map(|entry| {
    let entry = entry.ok()?;
    let path = entry.path();
    if !path.is_file() {
        return None;
    }
    let extension = path.extension()?.to_str()?;
    if !["mp3", "wav", "ogg", "flac"].contains(&extension.to_lowercase().as_str()) {
        return None;
    }
    Some(path)
})
```
The logic might not be correct, so be sure to verify